File: core/src/main/java/net/dontdrinkandroot/wicket/utils/NonStatelessPrintingVisitor.java
package net.dontdrinkandroot.wicket.utils;

import org.apache.wicket.Component;
import org.apache.wicket.util.visit.IVisit;
import org.apache.wicket.util.visit.IVisitor;

/**
 * Prints all components that are not stateless.
 *
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class NonStatelessPrintingVisitor implements IVisitor<Component, Void>
{
    @Override
    public void component(Component component, IVisit<Void> visit)
    {
        if (!component.isStateless()) {
            System.err.println(
                    String.format(
                            "Component '%s' is not stateless. Type: %s, Path: %s",
                            component.getMarkupId(),
                            component.getClass().getCanonicalName(),
                            component.getPath()
                    )
            );
        }
    }
}

File: core/src/main/kotlin/net/dontdrinkandroot/wicket/util/NonStatelessPrintingVisitor.kt
package net.dontdrinkandroot.wicket.util

import org.apache.wicket.Component
import org.apache.wicket.util.visit.IVisit
import org.apache.wicket.util.visit.IVisitor

/**
 * Prints all components that are not stateless.
 */
class NonStatelessPrintingVisitor : IVisitor<Component, Void> {

    override fun component(component: Component, visit: IVisit<Void>) {
        if (!component.isStateless) {
            System.err.println(
                String.format(
                    "Component '%s' is not stateless. Type: %s, Path: %s",
                    component.markupId,
                    component.javaClass.canonicalName,
                    component.path
                )
            )
        }
    }
}